Create.c

```diff
@@ -294,8 +294,8 @@ static int add_disk_to_super(int mdfd, struct shape *s, struct context *c,
 	}
 	if (fd >= 0)
 		remove_partitions(fd);
-	if (st->ss->add_to_super(st, &info->disk, fd, dv->devname,
-				 dv->data_offset)) {
+	if ((st->ss->add_to_super(st, &info->disk, fd, dv->devname,
+				 dv->data_offset)) && (!have_container)) {
```

Comment:
From review perspective I cannot judge if it is safe. Did you tested it?

I understand that we set `fd = -1` if `have_container` but it skipps `ioctl(mdfd, STOP_ARRAY, NULL);` is a logical change. Perhaps you can just use `close_fd(&fd)` and avoid this?